src/transaction/builder/mod.rs

```diff
@@ -9,313 +9,192 @@ use crate::{
-                        } => {
-                            if change_capacity < threshold {
-                                Self::add_output_capacity(&mut self.tx, target, change_capacity)?;
-                                state = BalanceState::Success;
-                                break;
-                            }
-                        }
-                        super::SmallChangeAction::AsFee { threshold } => {
-                            if change_capacity < threshold {
-                                state = BalanceState::Success;
-                                break;
-                            }
-                        }
-                    }
-                    {
-                        // init change
-                        let change_output = CellOutput::new_builder()
-                            .capacity(Capacity::bytes(0).unwrap().pack())
-                            .lock(self.change_lock.as_ref().unwrap().clone())
-                            .build();
-                        let change_output_data = packed::Bytes::default();
-                        mini_change_capacity = change_output
-                            .occupied_capacity(Capacity::bytes(change_output_data.len()).unwrap())
-                            .unwrap()
-                            .as_u64();
-                        self.change_output_index = Some(self.tx.get_outputs().len());
-                        self.tx.output(change_output);
-                        self.tx.output_data(change_output_data);
-                    }
-                    let new_fee = calculator.fee_with_tx_builder(&self.tx);
-                    if let Some(new_change) =
-                        (inputs_capacity + self.reward).checked_sub(outputs_capacity + new_fee)
-                    {
-                        change_capacity = new_change;
-                    } else {
-                        state = BalanceState::CapacityNotEnoughForChange(
-                            mini_change_capacity + new_fee - fee,
-                            change_capacity,
-                        );
-                        continue;
-                    }
-                }
-                if change_capacity >= mini_change_capacity {
-                    self.set_change_output_capacity(change_capacity);
-                    state = BalanceState::Success;
-                    break;
-                } else {
-                    state = BalanceState::CapacityNotEnoughForChange(
-                        mini_change_capacity,
-                        change_capacity,
-                    );
-                }
-            } else {
-                state = BalanceState::CapacityNotEnough(
-                    (outputs_capacity + fee) - (inputs_capacity + self.reward),
-                );
+            let input_capacity: u64 = previous_output.capacity().unpack();
+            inputs_capacity += input_capacity;
+            if inputs_capacity > required_capacity {
```

Comment:
if is possible that inputs_capacity equal required_capacity?

Reply:
it's possible, updated the pr, thanks.